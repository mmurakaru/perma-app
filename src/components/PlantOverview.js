import styled from 'styled-components/macro'

export default function PlantOverview({ plants }) {
    return (
        <ThumbnailWrapper>
            {plants?.map(({ id, common_name, image_url }) => (
                <PlantThumbnail key={id}>
                    <img src={image_url} alt="" />
                    <h2>{common_name}</h2>
                </PlantThumbnail>
            ))}
        </ThumbnailWrapper>
    )
}

const PlantThumbnail = styled.section`
    img {
        height: 100px;
        width: 156px;
        object-fit: cover;
        border-radius: 5px;
    }
`

const ThumbnailWrapper = styled.div`
    margin-top: 20px;
    height: 600px;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-auto-rows: min-content;
    place-items: center;
    gap: 20px;
    overflow: auto;
`
