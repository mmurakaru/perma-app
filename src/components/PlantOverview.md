PlantOverview example:

```js
<PlantOverview
    plants={[
        {
            id: 1,
            common_name: 'test',
            image_url: 'https://source.unsplash.com/random',
        },
        {
            id: 2,
            common_name: 'test2',
            image_url: 'https://source.unsplash.com/random',
        },
    ]}
/>
```
