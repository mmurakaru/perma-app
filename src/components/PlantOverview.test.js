import PlantOverview from './PlantOverview'
import { render } from '@testing-library/react'

describe('Overview', () => {
    it('renders correctly', () => {
        const { container } = render(
            <PlantOverview
                id={1}
                common_name="Strawberry"
                image_url="https://source.unsplash.com/random"
            />
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
