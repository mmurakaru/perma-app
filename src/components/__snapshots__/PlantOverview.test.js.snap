// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview renders correctly 1`] = `
<div
  class="PlantOverview__ThumbnailWrapper-dqgz7b-1 dKNswe"
/>
`;
